Nonreproducable Test Failure

### Description

`test_command_killthreads_produces_error_when_unknown_thread_passed` failed sporadically during some testing.

might be something to keep an eye on in case it happens again.

```
tests/test_command_killthreads.py::test_command_killthreads_produces_error_when_unknown_thread_passed 
Program stopped.
0x00007ffff7fe32b0 in _start () from /lib64/ld-linux-x86-64.so.2
Breakpoint 1 at 0x55555555528f: file multiple_threads.c, line 31.
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff7bff640 (LWP 31752)]
[New Thread 0x7ffff73fe640 (LWP 31753)]
[Thread 0x7ffff73fe640 (LWP 31753) exited]
[Thread 0x7ffff7bff640 (LWP 31752) exited]
[Inferior 1 (process 31629) exited with code 01]
FAILED

=================================== FAILURES ===================================
______ test_command_killthreads_produces_error_when_unknown_thread_passed ______

start_binary = <function start_binary.<locals>._start_binary at 0x7ff46241dfc0>

    def test_command_killthreads_produces_error_when_unknown_thread_passed(start_binary):
        start_binary(REFERENCE_BINARY_THREADS)
    
        gdb.execute("break break_here")
        gdb.execute("run")
        initial_thread_count = len(gdb.selected_inferior().threads())
        # check if thread with id 3 exists
>       assert len([thread for thread in gdb.selected_inferior().threads() if thread.num == 3]) == 1
E       assert 0 == 1
E        +  where 0 = len([])

initial_thread_count = 0
start_binary = <function start_binary.<locals>._start_binary at 0x7ff46241dfc0>

tests/test_command_killthreads.py:70: AssertionError
=========================== short test summary info ============================
FAILED tests/test_command_killthreads.py::test_command_killthreads_produces_error_when_unknown_thread_passed - assert 0 == 1
============================== 1 failed in 0.34s ===============================
```

### Steps to reproduce

CNR.

### My setup

```
pwndbg> version
Gdb:      12.1
Python:   3.10.6 (main, Mar 10 2023, 10:55:28) [GCC 11.3.0]
Pwndbg:   2023.03.19 build: 8a94940
Capstone: 4.0.1024
Unicorn:  2.0.1
```

Hmm:

```py
   def test_command_killthreads_produces_error_when_unknown_thread_passed(start_binary):
        start_binary(REFERENCE_BINARY_THREADS)
    
        gdb.execute("break break_here")
        gdb.execute("run")
        initial_thread_count = len(gdb.selected_inferior().threads())
        # check if thread with id 3 exists
>       assert len([thread for thread in gdb.selected_inferior().threads() if thread.num == 3]) == 1
E       assert 0 == 1
E        +  where 0 = len([])
```

We start [this program](https://github.com/pwndbg/pwndbg/blob/3016004a6c1fb37572925490dfa4fc57ada6d841/tests/gdb-tests/tests/binaries/multiple_threads.c) and break on `break_here` which is on the main thread and should be called only if we finished waiting for the two threads. Now, one of the thread executes `thread_function` which will sleep for 4 seconds and exit (!):
```c
void *thread_function(void *arg) {
    sem_post(&sem);
    usleep(4000);
    exit(1);
    return NULL;
}
...
    int res1 = pthread_create(&thread1, NULL, thread_function, NULL);
```

While the latter thread will wait forever:
```c
void *useless_thread(void *arg) {
    // signal the semaphore to indicate that the thread has started
    sem_post(&sem);

    while (1) {
        usleep(1000);
    }
    return NULL;
}

...
    int res2 = pthread_create(&thread2, NULL, useless_thread, NULL);
```

Afaik GitHub Actions have 2 cpu core and we use 3 threads here. Can it be that the 1st thread called exit before we check for threads? Would we get the same or a different error?

We never asserted/sanity checked `initial_thread_count = len(gdb.selected_inferior().threads())`, so maybe we did have 0 threads in the first place because the process exited?

--- a/tests/gdb-tests/tests/binaries/multiple_threads.c
+++ b/tests/gdb-tests/tests/binaries/multiple_threads.c
@@ -6,14 +6,6 @@
 
 sem_t sem;
 
-void *thread_function(void *arg)
-{
-    sem_post(&sem);
-    usleep(4000);
-    exit(1);
-    return NULL;
-}
-
 void *useless_thread(void *arg)
 {
     // signal the semaphore to indicate that the thread has started
@@ -37,7 +29,7 @@ int main()
     sem_init(&sem, 0, 0);
 
     pthread_t thread1, thread2;
-    int res1 = pthread_create(&thread1, NULL, thread_function, NULL);
+    int res1 = pthread_create(&thread1, NULL, useless_thread, NULL);
     if (res1 != 0) {
         perror("Thread creation failed");
         exit(EXIT_FAILURE);

--- a/tests/gdb-tests/tests/test_command_killthreads.py
+++ b/tests/gdb-tests/tests/test_command_killthreads.py
@@ -33,10 +33,6 @@ def test_command_killthreads_kills_all_threads_except_current(start_binary):
     # check if only one thread is left
     wait_until(lambda: len(gdb.selected_inferior().threads()) == 1)
 
-    gdb.execute("continue")
-    exit_code = gdb.execute("print $_exitcode", to_string=True)
-    assert exit_code == "$1 = 0\n"  # if the second thread was killed, the exit code should be 0
-
 
 def test_command_killthreads_kills_specific_thread(start_binary):
     start_binary(REFERENCE_BINARY_THREADS)

